### PR TITLE
TRUNK-5798, Add onDelete=CASCADE in liquibase changesets while creati…

### PR DESCRIPTION
**Pull request by Am-Coder**
…ng tables wherever necessary

## Description of what I changed

Added onDelete=CASCADE in liquibase schema files where necessary.
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5798

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by dkayiwa** [comment]
We are not touching the 2.3 database. We are doing all the postgresql changes starting from 2.4.0

**Comment by Am-Coder** [comment]
> We are not touching the 2.3 database. We are doing all the postgresql changes starting from 2.4.0

Understood, I have made some changes.

**Comment by Am-Coder** [comment]
@dkayiwa , I went through the hibernate config files for different entities. Only for the entities that I have added cascade in this PR have on-delete=CASCADE set in their hibernate configuration ( see [here](https://github.com/openmrs/openmrs-core/search?q=on-delete&unscoped_q=on-delete) ) . So, it appears that in OpenMRS we expect cascaded deletes at these places only. 
With the current changes in place we are able to successfully delete Patient forever from Patient Management section.

**Comment by dkayiwa** [comment]
Did you intentionally make this a draft pull request?

**Comment by Am-Coder** [comment]
Initially, I was not aware of exact number of changes required so I created a draft PR. But now I think its ready for review. Please, have a look at it.

**Comment by dkayiwa** [comment]
Does the command for integration tests succeed? `mvn clean test -Pskip-default-test -Pintegration-test`

**Comment by Am-Coder** [comment]
> Does the command for integration tests succeed? `mvn clean test -Pskip-default-test -Pintegration-test`

Yes, I just ran it and it is successful

File: api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.4.x.xml [hunk 1]
```diff
@@ -108,4 +108,37 @@
 			SELECT setval('order_type_order_type_id_seq', (SELECT MAX(order_type_id) FROM order_type)+1);
 		</sql>
 	</changeSet> 
+	
+	<changeSet id="20200622-cascade_delete_concept_comlex" author="aman" dbms="postgresql">
+		<preConditions onFail="MARK_RAN">
+    	    <foreignKeyConstraintExists foreignKeyName="concept_attributes"/>
+		</preConditions>
+		<comment>Updating Foreign Key To Add delete CASCADES</comment>
```
Comment (dkayiwa) [inline api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.4.x.xml:116]: Why do we have the same comment and yet the constraints are different?

File: api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.4.x.xml [hunk 2]
```diff
@@ -108,4 +108,37 @@
 			SELECT setval('order_type_order_type_id_seq', (SELECT MAX(order_type_id) FROM order_type)+1);
 		</sql>
 	</changeSet> 
+	
+	<changeSet id="20200622-cascade_delete_concept_comlex" author="aman" dbms="postgresql">
+		<preConditions onFail="MARK_RAN">
+    	    <foreignKeyConstraintExists foreignKeyName="concept_attributes"/>
+		</preConditions>
+		<comment>Updating Foreign Key concept_attributes To Add delete CASCADES</comment>
```
Comment (dkayiwa) [inline api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.4.x.xml:116]: Is there any reason why you capitalised `To Add`?

**Pull request merged**